global variables defined multiple times

https://stackoverflow.com/questions/14526153/multiple-definition-of-a-global-variable

I think that the solution on this stackoverflow post might be useful here.

Running /usr/lib/R/bin/R CMD INSTALL /tmp/RtmpUxnhj0/qtlbim_2.0.7.tar.gz --install-tests 
* installing to library ‘/home/fred/R/x86_64-pc-linux-gnu-library/4.0’
* installing *source* package ‘qtlbim’ ...
** using staged installation
** libs
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-8T8CYO/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c GlobalVars.c -o GlobalVars.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-8T8CYO/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c GlobalVars_SingleTrait.c -o GlobalVars_SingleTrait.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-8T8CYO/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c MatrixUtils.c -o MatrixUtils.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-8T8CYO/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c MultipleTraitsMCMC.c -o MultipleTraitsMCMC.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-8T8CYO/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RMultipleTraitsMCMCSetup.c -o RMultipleTraitsMCMCSetup.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-8T8CYO/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c ROutputManager.c -o ROutputManager.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-8T8CYO/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c RSingleTraitMCMCSetup.c -o RSingleTraitMCMCSetup.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-8T8CYO/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SingleTraitMCMC.c -o SingleTraitMCMC.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-8T8CYO/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c SingleTraitMCMCSamplingRoutines.c -o SingleTraitMCMCSamplingRoutines.o
gcc -std=gnu99 -I"/usr/share/R/include" -DNDEBUG      -fpic  -g -O2 -fdebug-prefix-map=/build/r-base-8T8CYO/r-base-4.0.3=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -g  -c StatUtils.c -o StatUtils.o
gcc -std=gnu99 -shared -L/usr/lib/R/lib -Wl,-Bsymbolic-functions -Wl,-z,relro -o qtlbim.so GlobalVars.o GlobalVars_SingleTrait.o MatrixUtils.o MultipleTraitsMCMC.o RMultipleTraitsMCMCSetup.o ROutputManager.o RSingleTraitMCMCSetup.o SingleTraitMCMC.o SingleTraitMCMCSamplingRoutines.o StatUtils.o -L/usr/lib/R/lib -lR
MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:34: multiple definition of `Y'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:77: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:73: multiple definition of `AMU'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:30: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:43: multiple definition of `GVALUE'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:79: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:56: multiple definition of `QCHR'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:13: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:52: multiple definition of `COEF'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:9: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:83: multiple definition of `MAIN'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:40: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:75: multiple definition of `GAMMA'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:32: first defined here
'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:54: first defined heredefinition of `GAMMA_EPISTASIS
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:93: multiple definition of `EPISTATIC'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:52: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:103: multiple definition of `GAMMA_GBYE'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:64: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:67: multiple definition of `GBYE_FIX'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:24: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:77: multiple definition of `VE'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:34: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:87: multiple definition of `VMAIN'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:44: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:97: multiple definition of `VEPISTASIS'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:56: first defined here
 defined hereMultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:105: multiple definition of `V_GBYE_FIX'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:66: first
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:61: multiple definition of `FIX'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:18: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:63: multiple definition of `RAN'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:20: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:65: multiple definition of `VRAN'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:22: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:54: multiple definition of `QLOC'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:11: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:26: multiple definition of `GENO'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:75: first defined here
/usr/bin/ld: MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:85: multiple definition of `GAMMA_MAIN'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:42: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:12: multiple definition of `QTLLOC'; MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:12: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:36: multiple definition of `NPHENO'; MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:36: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:38: multiple definition of `MULTIPLE'; MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:38: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:34: multiple definition of `Y'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:77: first defined here
eTraits.h:14: multiple definition of `DiffLocation'; MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:14: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:87: multiple definition of `VMAIN'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:44: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:97: multiple definition of `VEPISTASIS'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:56: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:105: multiple definition of `V_GBYE_FIX'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:66: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:73: multiple definition of `AMU'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:30: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:16: multiple definition of `SIGMA'; MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:16: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:83: multiple definition of `MAIN'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:40: first defined here
tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:52: first defined hereVars_MultipleTraits.h:93: multiple definition of `EPISTATIC'; GlobalVars_SingleTrait.o:/
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:43: multiple definition of `GVALUE'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:79: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:26: multiple definition of `GENO'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:75: first defined here
-usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:52: multiple definition of `COEF'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:9: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:54: multiple definition of `QLOC'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:11: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:56: multiple definition of `QCHR'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:13: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:75: multiple definition of `GAMMA'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:32: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:85: multiple definition of `GAMMA_MAIN'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:42: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:95: multiple definition of `GAMMA_EPISTASIS'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:54: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:61: multiple definition of `FIX'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:18: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:63: multiple definition of `RAN'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:20: first defined here
lbim/src/GlobalVars_MultipleTraits.h:65: multiple definition of `VRAN'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:22: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:67: multiple definition of `GBYE_FIX'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:24: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:103: multiple definition of `GAMMA_GBYE'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:64: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:77: multiple definition of `VE'; GlobalVars_SingleTrait.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_SingleTrait.c:34: first defined here
/usr/bin/ld: RMultipleTraitsMCMCSetup.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:18: multiple definition of `SCALE'; MultipleTraitsMCMC.o:/tmp/RtmpyQAn0K/R.INSTALL3e55394f648/qtlbim/src/GlobalVars_MultipleTraits.h:18: first defined here
collect2: error: ld returned 1 exit status
make: *** [/usr/share/R/share/make/shlib.mk:10: qtlbim.so] Error 1
ERROR: compilation failed for package ‘qtlbim’
* removing ‘/home/fred/R/x86_64-pc-linux-gnu-library/4.0/qtlbim’
Error in (function (command = NULL, args = character(), error_on_status = TRUE,  : 
  System command 'R' failed, exit status: 1, stdout & stderr were printed
Type .Last.error.trace to see where the error occured
Error in validate_signal_args(msg, type, call, .subclass) : 
  unused arguments (call, .subclass)

--- a/src/GlobalVars.c
+++ b/src/GlobalVars.c
@@ -117,3 +117,9 @@ char gbyefile[100];
 char covfile[100];
 char devfile[100];
 char sigmafile[100];
+
+double **SIGMA;
+int MULTIPLE;
+int NPHENO;
+int DiffLocation;
+int QTLLOC;

--- a/src/GlobalVars_MultipleTraits.h
+++ b/src/GlobalVars_MultipleTraits.h
@@ -9,13 +9,13 @@
 
 
 /// 0=uses QTLPOSITION_samelocation 1= uses QTLPOSITION
-int QTLLOC;
+extern int QTLLOC;
 /// Whether different locations are considered for each trait for each indicator or same locations are considered
-int DiffLocation;       
+extern int DiffLocation;
 /// Residual variance matrix for multiple triats
-double **SIGMA;
+extern double **SIGMA;
 /// Main Effect Variance scale for beta ~ N(0,cI) c=scale;
-double SCALE;
+extern double SCALE;
 //*******************************************************************************************
 
 
@@ -23,7 +23,7 @@ double SCALE;
 /// @{
 
 /// QTL genotypes for the MT code
-int ***GENO;
+extern int ***GENO;
 
 /// @}
 
@@ -31,16 +31,16 @@ int ***GENO;
 /// @{
 
 /// phenotypic data for the MT code
-double **Y;
+extern double **Y;
 /// The number of phenotypes to be jointly analyzed
-int NPHENO;            
+extern int NPHENO;
 /// 
-int MULTIPLE;
+extern int MULTIPLE;
 
 /// @}
 
 /// genotypic valyes for the MT code
-double **GVALUE;
+extern double **GVALUE;
 
 
 //***************************************************************************************
@@ -49,60 +49,60 @@ double **GVALUE;
 
 
 /// coefficients of QTL main effects
-double ****COEF;
+extern double ****COEF;
 /// QTL position indicators, position is GRID[QCHR[L]][QLOC[L]]
-int **QLOC;
+extern int **QLOC;
 /// chromosomes that QTL locate
-int **QCHR;
+extern int **QCHR;
 
     /// @defgroup qbOpCovariate_MT Multiple Traits variant of qb.op.covariate
     /// @{
     /// effects of fixed covariates
-    double  **FIX;
+    extern double  **FIX;
     /// effects of random covariates
-    double  ***RAN;
+    extern double  ***RAN;
     /// variances of random covariates
-    double  **VRAN;
+    extern double  **VRAN;
     /// interactions of QTL main effects and fixed covariates
-    double ****GBYE_FIX;
+    extern double ****GBYE_FIX;
     /// @}
 
     /// @defgroup qbOpIterdiag_MT Multiple Traits variant of qb.op.iterdiag
     /// @{
     /// overall mean
-    double *AMU;
+    extern double *AMU;
     /// QTL indicators
-    int **GAMMA;
+    extern int **GAMMA;
     /// residual variance
-    double VE;
+    extern double VE;
     /// @}
 
     /// @defgroup qbOpMainloci_MT Multiple Traits variant of qb.op.mainloci
     /// @{
     /// main effects
-    double  ***MAIN;
+    extern double  ***MAIN;
     /// main effects indicators
-    int **GAMMA_MAIN;
+    extern int **GAMMA_MAIN;
     /// prior variance of main effects
-    double  ***VMAIN;
+    extern double  ***VMAIN;
     /// @}
 
     /// @defgroup qbOpPairloci_MT Multiple Traits variant of qb.op.pairloci
     /// @{
     /// epistatic effects
-    double *****EPISTATIC;
+    extern double *****EPISTATIC;
     /// epistatic effects indicators
-    int ***GAMMA_EPISTASIS;
+    extern int ***GAMMA_EPISTASIS;
     /// prior variance of epistatic effects
-    double *****VEPISTASIS;
+    extern double *****VEPISTASIS;
     /// @}
 
     /// @defgroup qbOpGbyE_MT Multiple Traits variant of qb.op.gbye
     /// @{
     /// g-by-e indicators
-    double  ***GAMMA_GBYE;
+    extern double  ***GAMMA_GBYE;
     /// prior variance of g-by-e effects
-    double ****V_GBYE_FIX;
+    extern double ****V_GBYE_FIX;
     /// @}
 
 /// @}

--- a/src/Makevars
+++ b/src/Makevars
@@ -1,1 +1,0 @@
-PKG_CFLAGS = -fcommon

